### PR TITLE
Update OWNERS file with adding more approvers

### PR DESCRIPTION
**Pull request by jstourac**

### CODE REVIEW

**Review by harshad16 — COMMENTED** [review]
I believe we should first confirm with everyone if they all want this responsibility 
before including this PR.

/hold

File: CONTRIBUTING.md
Comment (jiridanek) [inline CONTRIBUTING.md:63]: **AI says**:

Sure, here is a gender-neutral and fluent version of the document:

```suggestion
### Review and Merge Process

- Once the PR is submitted, you can either select specific reviewers or let the bot to select reviewers [automatically](https://prow.ci.openshift.org/plugins?repo=opendatahub-io%2Fnotebooks).
- For the PR to be merged, it must receive 2 `/lgtm` comments from the repository [approvals/reviewers](/OWNERS). Following that, an `/approve` comment must be added by someone with approval rights. If the author of the PR has approval rights, it is preferred that they perform the merge action.
```

**Review by paulovmr — APPROVED** [review]

File: CONTRIBUTING.md
Comment (jstourac) [reply CONTRIBUTING.md]: thank you, I updated the text :slightly_smiling_face:

**Comment by jstourac** [comment]
At the moment, there are only the people who are explicitly okay with being approvers for this repository. Please, let me know in case anymore changes are necessary! :slightly_smiling_face:

**Review by caponetto — APPROVED** [review]

**Comment by atheo89** [comment]
It's always good to have new ideas for improvements but I am not fully convinced yet.

I feel that each step of our process should have only the appropriate people, it doent's sound correct to have the same people in two consecutive steps. I mean obviously that could cause a degradation. The owners are the people that know the component almost from end to end, and they have a knowledge history of it. 

I will bring an example,
Imagine to have a PR open from a contributor that re-introduce some things that are already covered by a feature long time ago, a member from our team likes this changes, imagine now another member that may review this pr and don't know the exact history assign also `lgtm`, so here, we can have two `lgtm` so any of these two people can allow a merge. In this scenario a merge could be fatal in our code base. 

Also, even if the unit tests would be fully matured (which in our case they aren't) we could not just rely solely on them for approvals etc... So, maybe temporarily we will gain some time but i am afraid that we loose time later on chasing unexpected issues.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve
/unhold

thanks for the PR :100: 
Kudos to the team for having more approvers

**Comment by jstourac** [comment]
Thank you, guys. I'll now backport the changes to other active branches in the upstream. Later in the downstream.

2024a seem to miss just this one commit, so lets do:

/cherrypick 2024a

**Pull request merged**
